src/topology/tietze_extension.lean

```diff
@@ -0,0 +1,282 @@
+             ... < g y - (b - c) : sub_lt_sub_right hc _
+             ... ≤ g y - dg y    : sub_le_sub_left (dgmem _).2 _ },
+        rcases ha.exists_between hay with ⟨_, ⟨x, rfl⟩, ha, hxy⟩,
+        exact ⟨x, xu, hxy.le, hyxu.le⟩ } },
+    { refine ⟨xl y, xu, _, hyxu.le⟩,
+      simp [dg0 (or.inr hc), hxl] } },
+end
+
+lemma exists_extension_forall_mem_of_closed_embedding (f : X →ᵇ ℝ) {s : set ℝ} {e : X → Y}
+  [hs : ord_connected s] (hf : ∀ x, f x ∈ s) (hne : s.nonempty) (he : closed_embedding e) :
+  ∃ g : Y →ᵇ ℝ, (∀ y, g y ∈ s) ∧ g ∘ e = f :=
+begin
+  casesI is_empty_or_nonempty X,
+  { rcases hne with ⟨c, hc⟩,
+    refine ⟨const Y c, λ y, hc, funext $ λ x, is_empty_elim x⟩ },
+  rcases exists_extension_forall_exists_le_ge_of_closed_embedding f he with ⟨g, hg, hgf⟩,
+  refine ⟨g, λ y, _, hgf⟩,
+  rcases hg y with ⟨xl, xu, h⟩,
+  exact hs.out (hf _) (hf _) h
+end
+
+end bounded_continuous_function
+
+namespace continuous_map
+
+lemma exists_extension_forall_mem_of_closed_embedding (f : C(X, ℝ)) {s : set ℝ} {e : X → Y}
+  [hs : ord_connected s] (hf : ∀ x, f x ∈ s) (hne : s.nonempty) (he : closed_embedding e) :
+  ∃ g : C(Y, ℝ), (∀ y, g y ∈ s) ∧ g ∘ e = f :=
+begin
+  have h : ℝ ≃o Ioo (-1 : ℝ) 1 := order_iso_Ioo_neg_one_one ℝ,
+  set F : X →ᵇ ℝ :=
+  { to_fun := coe ∘ (h ∘ f),
+    continuous_to_fun := continuous_subtype_coe.comp (h.continuous.comp f.continuous),
+    bounded' := bounded_range_iff.1 ((bounded_Ioo (-1 : ℝ) 1).mono $
+      forall_range_iff.2 $ λ x, (h (f x)).2) },
+  set t : set ℝ := (coe ∘ h) '' s,
+  have ht_sub : t ⊆ Ioo (-1 : ℝ) 1 := image_subset_iff.2 (λ x hx, (h x).2),
+  haveI : ord_connected t,
+  { constructor, rintros _ ⟨x, hx, rfl⟩ _ ⟨y, hy, rfl⟩ z hz,
+    lift z to Ioo (-1 : ℝ) 1 using (Icc_subset_Ioo (h x).2.1 (h y).2.2 hz),
+    change z ∈ Icc (h x) (h y) at hz, rw [← h.image_Icc] at hz,
+    rcases hz with ⟨z, hz, rfl⟩,
+    exact ⟨z, hs.out hx hy hz, rfl⟩ },
+  have hFt : ∀ x, F x ∈ t, from λ x, mem_image_of_mem _ (hf x),
+  rcases F.exists_extension_forall_mem_of_closed_embedding hFt (hne.image _) he
+    with ⟨G, hG, hGF⟩,
+  set g : C(Y, ℝ) := ⟨h.symm ∘ cod_restrict G _ (λ y, ht_sub (hG y)), h.symm.continuous.comp $
+    continuous_subtype_mk _ G.continuous⟩,
+  have hgG : ∀ {y a}, g y = a ↔ G y = h a,
+    from λ y a, h.to_equiv.symm_apply_eq.trans subtype.ext_iff,
+  refine ⟨g, λ y, _, _⟩,
+  { rcases hG y with ⟨a, ha, hay⟩,
+    convert ha,
+    exact hgG.2 hay.symm },
+  { ext x, exact hgG.2 (congr_fun hGF _) }
+end
+
+lemma exists_extension_of_closed_embedding (f : C(X, ℝ)) (e : X → Y) (he : closed_embedding e) :
+  ∃ g : C(Y, ℝ), g ∘ e = f :=
+(exists_extension_forall_mem_of_closed_embedding f (λ x, mem_univ _) univ_nonempty he).imp $
+  λ g, and.right
+
+lemma exists_restrict_eq_forall_mem_of_closed {s : set Y} (f : C(s, ℝ)) {t : set ℝ}
+  [ord_connected t] (ht : ∀ x, f x ∈ t) (hne : t.nonempty) (hs : is_closed s) :
+  ∃ g : C(Y, ℝ), (∀ y, g y ∈ t) ∧ g.restrict s = f :=
```

Comment:
Is a variant something like
```lean
  ∃ g : C(Y, t), (g.restrict s).comp coe = f :=
```
useful, or even possible to state?

Reply:
It is possible (though it's `coe ∘ g.restrict s`, not the other way). The most natural way is to define `continuous_map.cod_restrict`. I don't want to do it till someone needs it.